### PR TITLE
[MRG+1] Rewritten two decoding examples for adaptation to Decoder object:

### PR DESCRIPTION
**Pull request by tbng**
* Which are: `examples/02_decoding/plot_haxby_anova_svm.py` and `examples/02_decoding/plot_haxby_grid_search.py`
* Updated the corresponding .rst files in doc/

Issue(s) related: solved partially gh-2415.
PR related: as division of tasks from gh-2476.

### CODE REVIEW

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (tbng) [inline examples/02_decoding/plot_haxby_grid_search.py:116]: I would love to have reviews & possible suggestions on this part as the doing the nested CV loop like this is not the most efficient. Scikit-learn `GridSearchCV` and `cross_val_score` cannot be used here since `decoder only take input in the form of Nifty Images file-path/format.

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (bthirion) [inline examples/02_decoding/plot_haxby_grid_search.py:4]: :-)

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (bthirion) [inline examples/02_decoding/plot_haxby_grid_search.py:89]:
```suggestion
screening_percentile_range = [1.25, 2.5, 3.75, 7.5, 12.5, 25.0]
```

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (bthirion) [reply examples/02_decoding/plot_haxby_grid_search.py:116]: You mean, we need change the API to supported nested CV ?

I think I'd rather keep it as is for that release and open an issue to revisit that later.

**Review by bthirion — COMMENTED** [review]
Minor details. Thx !

**Comment by tbng** [comment]
@bthirion It seems there are some overlaps with gh-2476 so after IRL discussion with @thomasbazeille I will wait for that one to get merged before fix the overlaps of this PR.

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (thomasbazeille) [inline examples/02_decoding/plot_haxby_grid_search.py:121]: I think :
```for sp in sp_range:
        decoder = Decoder(estimator='svc', mask=mask_filename,
                          smoothing_fwhm=4, cv=3, standardize=True,
                          screening_percentile=sp)
        decoder.fit(index_img(fmri_niimgs, train), y_train)
        scores.append(decoder.cv_scores)
```
is enough. It's not a proper cross-validation per se but it's enough to compare the impact of screening_percentage and it avoids calling sklearn functions which is not the goal of the examples I guess. Any idea @jeromedockes ?

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (thomasbazeille) [reply examples/02_decoding/plot_haxby_grid_search.py:116]: I think binh thinks reimplementing a cross-val here is heavy and I agree. I proposed a compromise below

Comment (thomasbazeille) [reply examples/02_decoding/plot_haxby_grid_search.py:116]: Actually @tbng I just noticed the section above. Maybe it's enough for an example, keeping the plotting section that come last . So I suggest removing the "Nested cross val" section.

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (thomasbazeille) [inline examples/02_decoding/plot_haxby_grid_search.py:62]: Maybe you could also demonstrate very quickly how to grid-search the SVC parameter (or for another model)  just to show the API of grid_param, and return the best param kept ?

File: examples/02_decoding/plot_haxby_anova_svm.py
Comment (thomasbazeille) [inline examples/02_decoding/plot_haxby_anova_svm.py:93]: Could you point to the decoding_intro docpage in a topic that explains the cross-validation process ?

**Comment by tbng** [comment]
@GaelVaroquaux not strictly related to this PR but the current merge in main branch fails because of a conflict in link references between PR gh-2504 and gh-2476
```diff
9 files changed, 732 insertions(+), 457 deletions(-)
doc/decoding/decoding_intro.rst                    | 444 +++++++--------------
doc/decoding/estimator_choice.rst                  | 202 ++++++----
doc/decoding/going_further.rst                     | 166 ++++++++
doc/decoding/index.rst                             |   1 +
doc/themes/nilearn/static/nature.css_t             |   5 +-
.../plot_haxby_glm_decoding.py}                    | 137 +++----
examples/04_glm_first_level/README.txt             |   5 +-
examples/05_glm_second_level/README.txt            |   4 +-
.../07_advanced/plot_advanced_decoding_scikit.py   | 225 +++++++++++

renamed    examples/07_advanced/plot_haxby_block_classification.py -> examples/02_decoding/plot_haxby_glm_decoding.py
```

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (bthirion) [reply examples/02_decoding/plot_haxby_grid_search.py:129]: So, we agree to keep this as-is and leave enhancements for a future PR ?

**Review by bthirion — APPROVED** [review]

**Comment by thomasbazeille** [comment]
can you merge master please, new PRs have been merged since last CI run

**Comment by thomasbazeille** [comment]
Seems ready for me, I think we should merge this quickly and open additional PRs if more changes are needed

**Comment by bthirion** [comment]
Any other review ? Will merge tomorrow unless I hear otherwise.

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (KamalakerDadi) [reply examples/02_decoding/plot_haxby_grid_search.py:129]: +1 for leaving it to future ```Decoder``` object enhancements.

**Review by KamalakerDadi — COMMENTED** [review]
See if you have time.

Overall, great work on the simplification of examples. 

If it is Ok, I would prefer to save Nifti image like objects in compressed format (.nii.gz)

Great job!

File: examples/02_decoding/plot_haxby_anova_svm.py
Comment (tbng) [reply examples/02_decoding/plot_haxby_anova_svm.py:37]: So you suggest changing the variable name from `func_filename` --> `func_img`

File: examples/02_decoding/plot_haxby_anova_svm.py
Comment (tbng) [reply examples/02_decoding/plot_haxby_anova_svm.py:51]: Yes you're correct, I can quickly fix this

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (tbng) [reply examples/02_decoding/plot_haxby_grid_search.py:102]: The `screening_percentile` is not that flexible compare to specifying a fix number of voxels because `screening_percentile` must be an integer between `[1, 100]`.

For example if the number of voxels is 100,000 we cannot choose less than 1000 voxels (1 percent) with `screening_percentile`

File: examples/02_decoding/plot_haxby_anova_svm.py
Comment (KamalakerDadi) [reply examples/02_decoding/plot_haxby_anova_svm.py:37]: Yes, but see if it makes sense to you.

File: examples/02_decoding/plot_haxby_anova_svm.py
Comment (KamalakerDadi) [reply examples/02_decoding/plot_haxby_anova_svm.py:51]: Thanks.

File: examples/02_decoding/plot_haxby_grid_search.py
Comment (KamalakerDadi) [reply examples/02_decoding/plot_haxby_grid_search.py:102]: Ok. Thanks for the details. I missed this thinking that screening percentile is passed to ANOVA.

Comment (thomasbazeille) [reply examples/02_decoding/plot_haxby_grid_search.py:102]: I think more flexibility with screening percentile is not a priority right now, we should leave it to future enhancements if needed by someone

Comment (KamalakerDadi) [reply examples/02_decoding/plot_haxby_grid_search.py:102]: yes i understand

**Comment by thomasbazeille** [comment]
3 approvals, CI passing and this resolves current CI fail, I'll merge. More improvements can be planned in future PRs

**Comment by GaelVaroquaux** [comment]
> 3 approvals, CI passing and this resolves current CI fail,

Thanks a lot!

We need to be careful not to push the bar too high in terms of merge:
this was a clear merge. We don't need more approvals (possibly less).

**Pull request merged**
